### PR TITLE
Cleanups for accepting scm-staging bot submissions

### PR DESCRIPTION
**Pull request by dirkmueller**
* Ignore .gitattributes in checkout
* Skip Name: check when it is a scm-staging bot submission (relies on build service API that is broken for git packages)
* Skip devel-group-maintainer check

### CODE REVIEW

File: check_source.py [hunk 1]
```diff
@@ -425,6 +431,14 @@ def check_spec_policy(self, old, directory, specs):
 
         return True
 
+    def source_is_scm_staging_submission(self, source_project):
+        """Checks whether the source project is a scm_submission source project"""
+
+        if not len(self.allowed_scm_submission_sources):
+            return False
```
[the remark this answers is not in the document]
Comment (dirkmueller) [reply check_source.py:438]: removed

File: check_source.py [hunk 2]
```diff
@@ -192,7 +194,7 @@ def check_source_submission(self, source_project, source_package, source_revisio
                 if match:
                     inair_renamed = target_package != match.group(1)
 
-        if not self.source_has_correct_maintainers(source_project):
+        if not self.source_is_scm_staging_submission(source_project) and not self.source_has_correct_maintainers(source_project):
```
[the remark this answers is not in the document]
Comment (Vogtinator) [reply check_source.py:197]: Answering myself: Nope

[the remark this answers is not in the document]
Comment (dirkmueller) [reply check_source.py:197]: this is about checking whether the "factory-maintainers" group is added to the devel project. but for scm-staging we have temporarily created projects, so right now it isn't adding factory-maintainers group. 

I will fix the bot to add the group and then we can revert this part. will add a todo statement

File: check_source.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dirkmueller) [reply check_source.py]: Actually the problem is that the default is `['']` so I do need to have that (as shown by tests)

File: check_source.py [hunk 3]
```diff
@@ -62,7 +63,7 @@ def target_project_config(self, project):
         self.allow_valid_source_origin = str2bool(config.get('check-source-allow-valid-source-origin', 'False'))
         self.valid_source_origins = set(config.get('check-source-valid-source-origins', '').split(' '))
         self.add_devel_project_review = str2bool(config.get('check-source-add-devel-project-review', 'False'))
-        self.allowed_scm_submission_sources = set(config.get('allowed-scm-submission-sources', '').split(' '))
+        self.allowed_scm_submission_sources = config.get('allowed-scm-submission-sources', '').split(' ')
```
Comment (Vogtinator) [inline check_source.py:66]:
```suggestion
        self.allowed_scm_submission_sources = config.get('allowed-scm-submission-sources', '').split()
```

File: check_source.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Vogtinator) [reply check_source.py:445]: The splitting was done incorrectly, see comment above

File: check_source.py (hunk 3 above)
Comment (dirkmueller) [reply check_source.py]: Nice!  included

File: check_source.py [hunk 4]
```diff
@@ -229,20 +236,25 @@ def check_source_submission(self, source_project, source_package, source_revisio
...
         filename = new_info.get('filename', '')
         expected_name = target_package
         if filename == '_preinstallimage':
             expected_name = 'preinstallimage'
-        if not (filename.endswith('.kiwi') or filename == 'Dockerfile') and new_info['name'] != expected_name:
-            shutil.rmtree(dir)
-            self.review_messages['declined'] = "A package submitted as %s has to build as 'Name: %s' - found Name '%s'" % (
-                target_package, expected_name, new_info['name'])
-            return False
+        # TODO(dmllr): self.package_source_parse cannot handle scm_sync submissions, so skip the check for now
```
Comment (Vogtinator) [inline check_source.py:251]: In which way does it fail?

**Review by Vogtinator — APPROVED** [review]
Changes look good, just the commit title is wrong now - this is more than cleanups

**Pull request merged**